libuuu/sdp.cpp

```diff
@@ -239,6 +240,29 @@ SDPBootCmd::SDPBootCmd(char *p) : SDPCmdBase(p)
 	insert_param_info("-cleardcd", &m_clear_dcd, Param::Type::e_bool);
 	insert_param_info("-dcdaddr", &m_dcd_addr, Param::Type::e_uint32);
 	insert_param_info("-scanlimited", &m_scan_limited, Param::Type::e_uint64);
+	insert_param_info("-barebox", &m_barebox, Param::Type::e_bool);
+}
+
+int SDPBootCmd::load_barebox(CmdCtx *ctx)
+{
+	const ROM_INFO *rom = search_rom_info(ctx->m_config_item);
+
+	if (!rom)
+		return 0;
+
+	if (!(rom->flags & ROM_INFO_NEED_BAREBOX_FULL_IMAGE))
+		return 0;
+
```

Comment:
why need check ROM_INFO_NEED_BAREBOX_FULL_IMAGE?  you already have option -barebox-bl33

Reply:
This is for making it more straight forward for the users. The user only need to:

```
SDP: boot -barebox -f <barebox.img>
```

instead of:

```
SDP: boot -f <barebox.img>
SDP: write -ivt 0 -barebox-bl33 -f <barebox.img>
```

The 2nd form is far more complex to me since the user need to know the internals of `uuu` else no one would specify `-ivt 0` which is required to take the correct `uuu` code path.

Reply:
Furthermore the `ROM_INFO_NEED_BAREBOX_FULL_IMAGE` is only required for *i.MX8M* devices, since the loading for *i.MX6* is not that complex.

Reply:
> Furthermore the `ROM_INFO_NEED_BAREBOX_FULL_IMAGE` is only required for _i.MX8M_ devices. Loading _i.MX6_ images is not that complex since it requires only one file transfer.

Can you add comments about ROM_INFO_NEED_BAREBOX_FULL_IMAGE in code? And explain why only i.MX8M need, i.MX6 needn't?

Reply:
Of course I can do that. By code you mean this function here, right? Not the `rominfo.cpp`.

Reply:
Kept here in this function, since IMHO this is more useful for code readers. I hope this is okay for you.